server/publicapi/items/service.py

```diff
@@ -67,19 +62,169 @@ def get(self, req, lookup):
+            instances of Python's datetime.date
+
+        :raises BadParameterValueError:
+            * if any of the dates is not in the ISO 8601 format
+            * if any of the dates is set in the future
+            * if the start date is bigger than the end date
+        """
+        try:
+            start_date = self._parse_iso_date(request_params.get('start_date'))
+        except ValueError:
+            raise BadParameterValueError(
+                desc="Invalid parameter value (start_date)") from None
+
+        try:
+            end_date = self._parse_iso_date(request_params.get('end_date'))
+        except ValueError:
+            raise BadParameterValueError(
+                desc="Invalid parameter value (end_date)") from None
+
+        # disallow dates in the future
+        future_err_msg = (
+            "{} date ({}) must not be set in the future "
+            "(current server date: {})")
+        today = date.today()
+
+        if (start_date is not None) and (start_date > today):
+            raise BadParameterValueError(
+                desc=future_err_msg.format(
+                    'Start', start_date.isoformat(), today.isoformat()
+                )
+            )
+
+        if (end_date is not None) and (end_date > today):
+            raise BadParameterValueError(
+                desc=future_err_msg.format(
+                    'End', end_date.isoformat(), today.isoformat()
+                )
+            )
+
+        # make sure that the date range limits make sense
+        if (
+            (start_date is not None) and (end_date is not None) and
+            (start_date > end_date)
+        ):
+            # NOTE: we allow start_date == end_date (for specific date queries)
+            raise BadParameterValueError(
+                desc="Start date must not be greater than end date")
+
+        # set default date range values if missing
+        if end_date is None:
+            end_date = date.today()
+
+        if start_date is None:
+            start_date = end_date - timedelta(1)
+
+        return start_date, end_date
+
+    def _create_date_range_filter(self, start_date, end_date):
+        """Create a MongoDB date range query filter from the given dates.
+
+        If both the start date and the end date are None, an empty filter is
+        returned. The filtering is performed on the `versioncreated` field in
+        database.
+
+        :param start_date: the minimum version creation date (inclusive)
+        :type start_date: datetime.date or None
+        :param end_date: the maximum version creation date (inclusive)
+        :type end_date: datetime.date or None
+
+        :return: MongoDB date range filter (as a dictionary)
+        """
+        if (start_date is None) and (end_date is None):
+            return {}  # nothing to set for the date range filter
+
+        date_filter = {'versioncreated': {}}
+
+        if start_date is not None:
```

Comment:
If start_date and end_date are equal, the query will not return any response, since version created is a datetime.